openprovider/openprovider.php

```diff
@@ -2034,14 +2034,17 @@ private function compareTwoCustomerArrays(array $customer_one, array $customer_t
      * @param stdClass $service
      * @return string domain name. Return empty string if domain property not exists in $service
      */
-    private function getDomainNameFromService(stdClass $service) {
-        foreach ($service->fields as $field) {
-            if ($field->key == 'domain') {
-                return $field->value;
+    public function getServiceDomain(stdClass $service): string
+    {
+        if (isset($service->fields)) {
+            foreach ($service->fields as $field) {
+                if ($field->key == 'domain') {
+                    return $field->value;
+                }
             }
         }
 
-        return '';
+        return $this->getServiceName($service);
```

Comment:
Why do you return here service name instead of domain name or empty string?

Reply:
I took this code from documentation